File: Difficulty: Medium/Container With Most Water/container-with-most-water.java
class Solution {
    public int maxWater(int arr[]) {
        // Code Here
        int max = 0;
        int i = 0;
        int j = arr.length-1;
        
        while(i < j){
            int min = Math.min(arr[i],arr[j]);
            int w = j-i;
            max = Math.max(max,(min*w));
            if(arr[i] < arr[j]) i++;
            else j--;
        }
        
        return max;
    }
}

File: Difficulty: Medium/Container With Most Water/container-with-most-water.cpp
class Solution {
  public:
    int maxWater(vector<int> &arr) {
        // code here
        int n = arr.size();
        int maxWater = 0;
        int lp = 0, rp = n-1;
        int max_water = 0;
        while(lp<rp)
        {
            int w = rp - lp;
            int h = min(arr[rp], arr[lp]);
            int container = w * h;
            maxWater = max(maxWater, container);
            arr[lp] < arr[rp] ? lp++ : rp--;
        }
        return maxWater;
    }
};